lib/src/main/java/scal/io/liger/MainActivity.java

```diff
@@ -589,14 +602,26 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
 
                 if (c instanceof ClipCard) {
                     ClipCard cc = (ClipCard)c;
-                    MediaFile mf = new MediaFile(path, Constants.PHOTO);
-                    cc.saveMediaFile(mf);
 
-                    // SEEMS LIKE A REASONABLE TIME TO SAVE
-                    mStoryPathLibrary.save(true);
+                    // confirm mime type
+                    String mimeType = URLConnection.guessContentTypeFromName(path);
+
+                    Log.d(TAG, "onActivityResult, media type is " + mimeType);
+
+                    if (mimeType.startsWith(Constants.IMAGE)) {
 
-                    mCardAdapter.changeCard(cc);
-                    scrollRecyclerViewToCard(cc);
+                        MediaFile mf = new MediaFile(path, Constants.PHOTO); // TODO: switch over to "image" to match mime type?
+                        cc.saveMediaFile(mf);
+
+                        // SEEMS LIKE A REASONABLE TIME TO SAVE
+                        mStoryPathLibrary.save(true);
+
+                        mCardAdapter.changeCard(cc);
+                        scrollRecyclerViewToCard(cc);
+                    } else {
+                        Log.e(TAG, "onActivityResult, expected " + Constants.IMAGE + " but found " + mimeType);
```

Comment:
instead of silently logging this, we need to raise the issue to the user so they know why their new import just vanished